modules/c/examples/code_snippets/main.c

```diff
@@ -1078,3 +1197,392 @@ Placeholder for Date accessors.
+
+
+
+// DOCS NOTE:
+// Page=Data Sync >> Configuration
+// URL=https://docs.couchbase.com/couchbase-lite/current/c/replication.html#configuration-summary
+// This function is not pulled into docs en-bloc
+// it is a slightly more in-depth than the configurationsummary above
+// and the snippets within it are used individually or in sets
+// to illustrate specific points as required
+//
+static void docs_act_replication_config_section_snippets() {
+    CBLDatabase* db = kDatabase;
+    bool docs_example_resetRequired = false;
+    bool docs_example_ShowBasicAuth = false;
+    bool docs_example_ShowSessionAuth = false;
+
+    /*
+    * This requires Sync Gateway running with the following config, or equivalent:
+    *
+    * {
+    *     "log":["*"],
+    *     "databases": {
+    *         "db": {
+    *             "server":"walrus:",
+    *             "users": {
+    *                 "GUEST": {"disabled": false, "admin_channels": ["*"] }
+    *             }
+    *         }
+    *     }
+    * }
+    */
+
+    // tag::p2p-act-rep-func-full[]
+    // NOTE: No error handling, for brevity (see getting started)
+    // Note: Android emulator needs to use 10.0.2.2 for localhost (10.0.3.2 for GenyMotion)
+
+
+    // tag::sgw-act-rep-initialize[]
+    // Initialize the configuration object and set db target
+    CBLError err;
+    FLString url = FLSTR("ws://localhost:4984/db");
+    CBLEndpoint* target =
+        CBLEndpoint_CreateWithURL(url, &err); // <.>
+
+    CBLReplicatorConfiguration config;
+    memset(&config, 0, sizeof(CBLReplicatorConfiguration));
+    config.database = db;
+    config.endpoint = target; // <.>
+
+    // end::sgw-act-rep-initialize[]
+
+    //    tag::p2p-act-rep-config-type[]
+    config.replicatorType = kCBLReplicatorTypePull;
+    //    end::p2p-act-rep-config-type[]
+    //    tag::p2p-act-rep-config-cont[]
+    config.continuous = true;
+    //    end::p2p-act-rep-config-cont[]
+
+
+    // tag::replication-retry-config[]
+    // Configure replication retries
+    // tag::replication-set-heartbeat[]
+    config.heartbeat = 120; //  <.>
+    // end::replication-set-heartbeat[]
+    // tag::replication-set-maxattempts[]
+    config.maxAttempts = 20; //  <.>
+    // end::replication-set-maxattempts[]
+    // tag::replication-set-maxattemptwaittime[]
+    config.maxAttemptWaitTime = 600; //  <.>
+    // end::replication-set-maxattemptwaittime[]
+
+    // end::replication-retry-config[]
+
+    // tag::basic-authentication[]
+    // Configure Client Authentication to Basic Authentication
+    // Providing username and password credentials
+    if(docs_example_ShowBasicAuth) {
+        CBLAuthenticator* basicAuth =
+            CBLAuth_CreatePassword(FLSTR("username"),
+                                   FLSTR("passwd"));
+        config.authenticator = basicAuth; // <.>
+    }
+    // end::basic-authentication[]
+
+    // tag::session-authentication[]
+    if(docs_example_ShowSessionAuth) {
```

Comment:
What is this `if` condition for?  It won't make sense in the context of a snippet.

Reply:
Right. In some examples I want to show two or more ways of doing something, or show off a number of options.

I thought rather than just include them all-in-row -- which would look daft in practice -- I'd include them in some pseudo-conditional that may potentially mimic a viable scenario. I'll revert if you prefer.

Reply:
It would make sense if you were demoing this code to someone to show it function, but if this snippet is getting stuck into docs HTML the first question is going to be "What is docs_example_ShowSessionAuth and why is it here?"